### PR TITLE
d3 graph

### PR DESCRIPTION
**Pull request by vincerubinetti**
- makes sections collapsible; turns section headers into buttons to collapse sections
- makes d3 graph display of gene network (removes vega)
- includes max genes filter and min edge weight filter sliders
- includes physics simulation to detangle nodes
- includes view panning and zooming
- includes node dragging
- includes edge thickness and color gradient based on normalized weight
- includes graph controls to fit view and pin/unpin nodes
- includes auto-fitting feature to auto fit view when force simulation is running (turns off if user interacts)
- includes graph svg download
- fixes graph construction process to match old adage (fixes gene ranking)
- makes graph construction process much more efficient with memoization
- adds loading status display to gene network
- renames "Aux" component to "Right"

image

### CODE REVIEW

File: src/controller.js [hunk 1]
```diff
@@ -80,7 +80,7 @@ let Controller = ({
       getGeneEdges({
         modelId: selectedModel,
         geneIds: selectedGenes.map((selected) => selected.id),
-        limit: 100
+        limit: 999999
```
Comment (dongbohu) [inline src/controller.js]: This number may not be large enough if you want to load all edges from the backend. The `Number.MAX_SAFE_INTEGER`  constant may be better:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/MAX_SAFE_INTEGER

File: src/pages/genes/network/filters/index.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      value={nodeCutoff}
+      min={selectedNodes}
+      max={fullNodes}
+      step={1}
+      onChange={setNodeCutoff}
+    />
+    <Slider
+      title='Edge weight cutoff'
+      value={edgeWeightCutoff}
+      min={0.4}
```
Comment (dongbohu) [inline src/pages/genes/network/filters/index.js]: The minimum should be the `g2g_edge_cutoff` value of selected mlmodel, see:
https://py3-adage.greenelab.com/api/v1/model/

**Review by dongbohu — COMMENTED** [review]
Left some comments.

File: src/controller.js (hunk 1 above)
Comment (vincerubinetti) [reply src/controller.js]: That is the maximum safe javascript integer, but is there a max safe number for django too? Just want to make sure we don't exceed that.

Comment (dongbohu) [reply src/controller.js]: Good question. Let's use `9999999` (seven 9's instead of six) for now. Django doc doesn't mention the max int it supports.

Comment (vincerubinetti) [reply src/controller.js]: okay, incorporated comments

**Review by dongbohu — COMMENTED** [review]
One last comment: can you change the maximum weight cutoff value from "1" to "1.0"? It was asked by one of the users of previous version, but it was never changed. Thanks. After that change, please feel free to close this PR.

**Pull request merged**